completion_aggregator/aggregator.py

```diff
@@ -43,9 +43,10 @@ def perform_aggregation():
             )
             if stale.block_key is None:
                 stale_blocks[enrollment] = utils.BagOfHolding()
-            elif len(stale_blocks[enrollment]) <= MAX_KEYS_PER_TASK:
+            blocks = stale_blocks[enrollment]
+            if isinstance(blocks, utils.BagOfHolding) or len(blocks) <= MAX_KEYS_PER_TASK:
                 # We can stop adding once we have exceeded the maximum number of keys per task.
-                stale_blocks[enrollment].add(stale.block_key)
+                blocks.add(stale.block_key)
             if stale.force:
```

Comment:
This fixes an issue where we were calling `len(stale_blocks[enrollment])`, but `stale_blocks[enrollment]` was a `BagOfHolding` (basically a set-like object that claims to contain anything you ask it about), which doesn't implement `__len__`.  @pomegranited This is relevant to the work you reviewed.

Reply:
Yep.. [lament](https://github.com/open-craft/openedx-completion-aggregator/pull/10#discussion_r185992287)!

Reply:
I am not sure I follow the logic here. Which call to `len(stale_blocks[enrollment])` may result in `BagOfHolding.__len__()` being called? Also `if isinstance(blocks, utils.BagOfHolding)` is `True`, doesn't that mean that `state.block_key` is going to be `None`?